docs/architecture/psa-thread-safety.md

```diff
@@ -362,3 +362,27 @@ Since we only have simple mutexes, locking the same mutex from the same thread i
 Releasing the mutex before a function call might introduce race conditions. Therefore might not be practical to take the mutex in low level access functions. If functions like that don't take the mutex, they need to rely on the caller to take it for them. These functions will document that the caller is required to hold the mutex.
 
 To avoid performance degradation, functions must not start expensive operations (eg. doing cryptography) while holding the mutex.
+
+### Testing
```

Comment:
Minor note on terminology: this is a part of a _test strategy_, describing general rules for how to test stuff. It isn't a _test plan_, which lists the tests that will be performed. (We don't usually go to the detail of a test plan in Mbed TLS documentation. Our test plan is the `.data` files.)

A missing part of the test strategy is how we'll decide what multithreaded tests we'll add. A test plan would take that and deduce a precise list of functions and behaviors to test. Another missing part is how we test a function that needs to be thread-safe (i.e. generalizing from the `mbedtls_ctr_drbg_random` PoC). None of these need to be in the current iteration of the document, but we have to have some idea for a detailed task planning.